src/main/java/org/folio/utils/MoneyUtils.java

```diff
@@ -61,4 +61,18 @@ public static MonetaryAmount calculateTotalAmount(List<Transaction> transactions
   public static double calculateTotalAmountWithRounding(List<Transaction> transactions, CurrencyUnit currency) {
     return calculateTotalAmount(transactions, currency).with(Monetary.getDefaultRounding()).getNumber().doubleValue();
   }
+
+  public static MonetaryAmount ensureNonNegative(MonetaryAmount amount, String currency) {
+    MonetaryAmount zeroMonetaryAmount = Money.of(0, currency);
+    return amount.isGreaterThan(zeroMonetaryAmount) ? amount : zeroMonetaryAmount;
+  }
+
+  public static Money ensureNonNegative(Money amount, String currency) {
+    Money zeroMonetaryAmount = Money.of(0, currency);
```

Comment:
This can be implemented in a simpler way:
MonetaryFunctions.max().apply(amount, Money.zero(currency))

Reply:
simplified all methods